src/ExpressionParser.php

```diff
@@ -105,7 +110,7 @@ public function parseExpression($precedence = 0, $allowArrow = false, bool $depr
             } elseif (isset($op['callable'])) {
                 $expr = $op['callable']($this->parser, $expr);
             } else {
-                $expr1 = $this->parseExpression(self::OPERATOR_LEFT === $op['associativity'] ? $op['precedence'] + 1 : $op['precedence'], true);
+                $expr1 = $this->parseExpression(self::OPERATOR_LEFT === $op['associativity'] ? $op['precedence'] + 1 : $op['precedence']);
```

Comment:
what if this is called from a place where `$this->deprecationCheck` was set to `false` ? Should this be a place where we turn it on again ? Wouldn't this miss deprecation checks for operators used in an parenthesized expression in a larger expression ?

Reply:
Right, I've also updated this part.

Reply:
I found more problems, fixed in #4382